File: 658-find-k-closest-elements/find-k-closest-elements.py
class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
        i = 0
        j = len(arr) - k

        while (i < j):
            mid = (i + j) // 2

            if (x - arr[mid] > arr[mid + k] - x):
                i = mid + 1
            else:
                j = mid

        
        return arr[i: i + k]
        

File: 658-find-k-closest-elements/find-k-closest-elements.js
var findClosestElements = function(arr, k, x) {
    let i = 0
    let j = arr.length - k

    while (i < j) {
        let mid = Math.floor((i + j) / 2)
        if (x - arr[mid] > arr[mid + k] - x) {
            i = mid + 1
        }
        else j = mid
    }

    return arr.slice(i, i + k)
};